sevsnpmeasure/id_block.py

```diff
@@ -0,0 +1,162 @@
+#
+# Copyright 2022- IBM Inc. All rights reserved
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import hashlib
+import base64
+import ctypes
+import argparse
+import sys
+from ctypes import c_uint8
+from typing import List
+
+from cryptography.hazmat.primitives.asymmetric import ec,utils
+from cryptography.hazmat.primitives import serialization, hashes
+from cryptography.hazmat.backends import default_backend
+
+LaunchDigest = c_uint8 * 48
+ECParam = c_uint8 * 52
+ECSignature = c_uint8 * 512
+ECPubKey = c_uint8 * 1028
+
+DefaultIDs = bytes(0x20)
+DefaultVersion = 1
+DefaultGuestSVN = bytes(4)
+DefaultPolicy = 196608 # [0,0,3,0,0,0,0,0]
+DefaultKeyAlgo = 1
+CurveP384 = 2
+ECKeyLength = 1028
+ECSigLength = 512
+
+
+
+def main() -> int:
+    parser = argparse.ArgumentParser(prog='snp-create-id-block',
+                                     description='Calculate AMD SEV-SNP guest id block')
+    parser.add_argument('--measurement', metavar='VALUE', type=str, help='Guest launch measurement', default=None)
+    parser.add_argument('--idkey', metavar='PATH', help='id private key file')
+    parser.add_argument('--authorkey', metavar='PATH', help='author private key file')
+    args = parser.parse_args()
+
+    if args.idkey == None or args.authorkey == None:
+        parser.error("missing key files for id block")
+    ld = base64.b64decode(args.measurement)
+    measurement = snp_calc_id_block(ld,args.idkey,args.authorkey)
+
+    print(measurement)
```

Comment:
rename variable -- this is not a measurement.
also add spaces after commas in the argument list.